types/tests/Permissible.test.tsx

```diff
@@ -0,0 +1,22 @@
+import * as React from 'react'
+import { render, } from 'react-dom'
+import { Permissible, UserAndRequiredPermissions, } from '@brainhubeu/react-permissible'
+
+function callbackFunction({ userPermissions, requiredPermissions, }: UserAndRequiredPermissions){
+  console.log(`
+    react-permissible: Access Denied
+    userPermissions: ${userPermissions}
+    requiredPermissions: ${requiredPermissions}
+  `)
+}
+
+const AccessGranted = ({ message, }: { message: string, }) => <>AccessGranted {message}</>
+
+const RestrictedComponentWithCallback = Permissible(
+  AccessGranted,
+  ['ACCESS_DASHBOARD'], // userPermissions
+  ['ACCESS_ADMIN'], // requiredPermissions
```

Comment:
I think it's good to leave the comments, as they describe what parameters are given to the HOC. 

At some point, I want to make the params consolidated to one object with named keys. Then we will remove the comments.

Reply:
Yeah, I follow this rule: whenever a function has more than one parameter (unless it is a binary function, such as a comparator), it should use a single object as an argument.
That's because most likely, at some point, two arguments will become three then four then all hell breaks loose :smiling_imp:.

Reply:
Yeah, that's what I think as well, didn't have time for fixing that, though 😎